spec/harvesting/models/users_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'spec_helper'
+
+RSpec.describe Harvesting::Models::Users, :vcr do
+  let(:attrs) { Hash.new }
+  let(:users) { Harvesting::Models::Users.new(attrs, {}, client: harvest_client) }
+
+  include_context "harvest data setup"
+
+  let(:page1_results) do
+    entries = []
+    100.times { entries.push({}) }
+    {
+        "users" => entries,
+        "per_page" => 100,
+        "total_pages" => 2,
+        "total_entries" => 115,
+        "next_page" => 2,
+        "previous_page" => nil,
+        "page" => 1
+    }
+  end
+
+  let(:page2_results) do
+    entries = []
+    15.times { entries.push({}) }
+    {
+        users: entries,
+        per_page: 100,
+        total_pages: 2,
+        total_entries: 115,
+        next_page: nil,
+        previous_page: 1,
+        page: 2
+    }
+  end
+
+  let(:attrs) { page1_results }
+
+  describe "#fetch_next_page" do
+    it "provides next page content" do
+      stub_request(:get, /users/).
+            with(query: {"page" => "2"}).
+            to_return( { body: page2_results.to_json } )
+
+      expect(users.entries.count).to eq(100)
+      expect(users.page).to eq(1)
+      users.send(:fetch_next_page)
```

Comment:
@chimedpalden I think it's best to not know so much about the implementation of the pagination logic... 

Could you change this scenario to iterate through `user.entries` 115 times? That should do what you expect it to do.